spec/win32/unit/store/assertions_spec.rb

```diff
@@ -0,0 +1,88 @@
+#
+# Author:: Piyush Awasthi (<piyush.awasthi@msystechnologies.com>)
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe Win32::Mixin::Assertions do
+
+  class Store
+    include Win32::Mixin::Assertions
+  end
+
+  let (:certstore) { Store.new }
+  
+  describe "#validate_store" do
+    context "When passing empty certificate store name" do
+      let (:store_name) { "" }
+      it "Raise ArgumentError" do
+        expect { certstore.validate_store(store_name) }.to raise_error(ArgumentError)
+      end
+    end
+
+    context "When passing invalid certificate store name" do
+      let (:store_name) { "Chef" }
+      it "Raise ArgumentError" do
+        expect { certstore.validate_store(store_name) }.to raise_error(ArgumentError)
```

Comment:
If we are manually raising some error anywhere, we should check with the actual error message instead of `ArgumentError`